### PR TITLE
Add option to preserve header

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR changes `transform` tranformation config to preserve markdown headers by default while transforming as discussed in 1:1. `popHeader` option may be added to enable removing header. 

Note: Could not find a better explicit way of enabling remove header other than bool option in config (as was mentioned in 1:1). Would a flag instead be preferred?

### CODE REVIEW

**Review by bwplotka — APPROVED** [review]
Nice LGTM, just one idea: Wonder if we can global default `popHeader` on root file? 🤔

**Comment by saswatamcode** [comment]
Now, `popHeader` will be true by default for root files in input dir. This can be disabled by setting `popHeader` to false with glob matching root file. Also, all root files will have this as true by default, so would need to be configured separately if this behavior is not needed. Test case added in `pkg/transform/testdata/mdox3.yaml` where `popHeader` is not set for `README.md` but md header is still popped.

File: pkg/transform/testdata/mdox1.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ gitIgnored: true
 transformations:
   - glob: "README.md"
     path: _index.md
+    popHeader: true
```
Comment (bwplotka) [inline pkg/transform/testdata/mdox1.yaml:11]: If popHeader is true on root, why we do this? 🤔

Comment (saswatamcode) [reply pkg/transform/testdata/mdox1.yaml]: Sorry, forgot to remove!

Comment (bwplotka) [reply pkg/transform/testdata/mdox1.yaml]: It's still there on all examples 🤔

**Comment by bwplotka** [comment]
Merging, LGTM thanks, but examples are  still not perfect 🙈

**Comment by saswatamcode** [comment]
I've missed some. Will raise PR to fix them! 🙂

**Pull request merged**